examples/tv-app/linux/include/application-basic/ApplicationBasicManager.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ApplicationBasicManager.h"
+#include "../endpoint-configuration/EndpointConfiguration.h"
+
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+#include "gen/command-id.h"
+
+#include <app/util/af.h>
+#include <app/util/basic-types.h>
+
+#include <inipp/inipp.h>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+using namespace std;
+
+void logApp(const Application & a)
+{
+    std::cout << "------------ PRINT -----------" << std::endl;
+    std::printf("catalogVendorId: %d\n", a.catalogVendorId);
+    std::cout << "id: " << a.id << std::endl;
+    std::cout << "name: " << a.name << std::endl;
+    std::printf("productId: %d\n", a.productId);
+    std::printf("vendorId: %d\n", a.vendorId);
+    std::cout << "vendorName: " << a.vendorName << std::endl;
+}
+
+char * convertToChar(string appString)
+{
+    char * cstr = new char[appString.length() + 1];
+    strcpy(cstr, appString.c_str());
+    return cstr;
+}
+
+int ApplicationBasicManager::Init()
+{
+    return 0;
+}
+
+EmberAfStatus ApplicationBasicManager::store(chip::EndpointId endpoint, Application * application)
+{
+    if (strcmp(convertToChar(application->vendorName), "\0") != 0)
+    {
+#ifndef NDEBUG
+        EmberAfStatus vendorNameStatus =
```

Comment:
this method already returns an EmberAfStatus.
should we not return the failure status if any of the failures below happen? 

Otherwise even on error, we still say OK .. this would be super painful to debug.

Reply:
Simplified. Method is void and only assert is used.